docs/OVERVIEW.md

```diff
@@ -0,0 +1,78 @@
+## Introduction
+
+### What are Confidential Identities?
+
+Confidential identities in Corda provide a way to restrict the knowledge of identities within a transaction to a need-to-know basis. In a normal transaction, the participants are `Party` objects which contain the `CordaX500Name` and the identity owning `PublicKey`. Confidential identities are modelled using `AnonymousParty` objects which only contain the owning `PublicKey`. 
+
+This mechanism allows a node to carry out transactions in which the participants within it do not want to be known to the network. Only the parties involved in the transaction will be able to resolve the confidential owning key. This CorDapp provides two flows; one for generating confidential identities - `RequestKeyFlow`, and one for sharing them with another party `SyncKeyMappingFlow`.
+
+This version of confidential identities differs from the old version in that we do not store the X.509 certificates for each confidential identity. This greatly reduces the storage overhead for systems that require large quantities of confidential identities. 
+
+### Generation of Confidential Identities
+
+#### RequestKeyFlow
+
+This process is provided entirely by calling `RequestKey` as an inline subflow. The various steps of the security protocol are detailed below.
+
+* Alice requests a new confidential identity from Bob
+
+
+   `aliceNode.subFlow(RequestKey(bob))`
+* Alice generates a new random `SHA256` value and sends this to Bob along with a request to generate a new key pair
+* Bob's `KeyManagementService` generates a new key pair and stores it 
+    
+ `keyManagementService.freshKey()`
+
+* Bob registers a mapping in his `IdentityService` between `newKeyForBob` and himself
+* Bob generates his own random `SHA256` value and concatenates it with the one alice sent
+* Bob signs over the concatenated value using the new `PublicKey`
+* Bob then sends the new `PublicKey`, the signed concatenated `SHA256` value and his own `SHA256` value back to Alice
+* Alice then verfies two things:
+  * The `PublicKey` matches that of the one used to sign over the concatenated `SHA256` values
+  * That the decrypted concatenated value matches the same value as when she concatenates her `SHA256` and the one Bob sent 
+
+* If the above criteria is fulfilled, Alice can be assured that the key came from Bob and will store a mapping between the new `PublicKey` and Bob in here `IdentityService`
+
+ ` serviceHub.identityService.registerKey(newKeyForBob, counterParty)`
```

Comment:
maybe note that you can optionally assign it to a UUID as well (if you are using accounts)